config/file_config.go

```diff
@@ -477,13 +477,28 @@ func (f *fileConfig) GetCompressPeerCommunication() bool {
 	f.mux.RLock()
 	defer f.mux.RUnlock()
 
-	return f.mainConfig.Specialized.CompressPeerCommunication
+	var compressPeerCommunication bool
+	if f.mainConfig.Specialized.CompressPeerCommunication == nil {
+		compressPeerCommunication = true
```

Comment:
These true values are hardcoded, but we should try and lookup these values from the struct tag. That will require us to use the reflect package mimicking some behavior from the defaults package. Is this something that we want in the scope of this fix?

Reply:
I'm not sure I'm happy with the hardcoded `true` values that are only present on struct values that are currently type `bool` and default `true`. It might be nice to write a helper function that we can add to add struct values to check is type is `bool` and default is `true`.

Reply:
@robbkidd suggested that we either add some code to the validate function to check for struct tag type and value. Or perhaps there is a test that we can add later on to validate any changes to struct tags.

Reply:
Instead of reading the tags, we talked about something like:

```
type BoolDefaultTrue *bool
```

That would give us the hook to do some more useful linting or other work to make sure that we keep the contract for those types.